src/Subsystem/ElevatorSubsytem/Elevator.java

```diff
@@ -108,33 +115,131 @@ private void sendStateUpdate(){
         transmitterToScheduler.send(stateEvent);
     }
 
-    /**
-     * Process the given event, identify type and select an action or activity based on it.
-     *
-     * @param event The event to process
-     * @throws InvalidTypeException If it receives an event type this class cannot handle
-     */
-    private void processMessage(SystemMessage event) throws InvalidTypeException {
-        // Note: Cannot switch on type, if we want to refactor selection, look into visitor pattern.
-        if (event instanceof MoveElevatorCommand)
-            move(((MoveElevatorCommand) event).direction());
-        else if (event instanceof MovePassengersCommand) {
-            unload();
-            load((MovePassengersCommand) event);
-        }
-        else // Default, should never happen
-            throw new InvalidTypeException("Event type received cannot be handled by this subsystem.");
-    }
-
     @Override
     public void run() {
+        // Set initial state.
+        //
+        // Decision to set initial state in method run because the elevator
+        // has no state until method run is invoked.
+        //
+        // Alternative would be setting initial state in constructor but this
+        // causes funny business with transmitters and receivers.
+        setState(new ReceivingState(this));
+
         while (true){
-            sendStateUpdate();
-            try {
-                processMessage(receiver.dequeueMessage());
-            } catch (InvalidTypeException e) {
+            // Wait until an event occurs in this state machine
+            event = receiver.dequeueMessage();
+
+            // Handle the event occurring in this state machine
+            if (event instanceof MoveElevatorCommand) {
+                ((ElevatorState) state).handleMoveElevatorCommand();
+            } else if (event instanceof MovePassengersCommand) {
+                ((ElevatorState) state).handleMovePassengersCommand();
+            } else {
+                InvalidTypeException e = new InvalidTypeException("Event type received cannot be handled by this subsystem.");
```

Comment:
the elevator should do any processing in the elevator other than basically transition between states. this run method shouldnt have the logic of processing events. this should be in the receiving state.

Reply:
having a runstate() method in the states where it runs entry, doactivity,exit would take care of all the logic. the elevator would only be responsible of starting the runstate for each state, delegating the actual processing to the state as much as possible.

Reply:
I'm sorry, I disagree with comment 1. 

If we are attempting to implement Elevator as a State Machine, then run() SHOULD have the logic of *handling* events. My implementation doesn't *process* the events. Processing is done in the various handler methods (eg. handleMoveElevatorCommand)

Addressing comment 2. Method changeState() is synonymous with runState().